client/blocks/login/social.jsx

```diff
@@ -55,6 +61,8 @@ class SocialLoginForm extends Component {
 									error_message: createAccountError.message
 								} )
 							);
+					} else if ( error.code === 'existing_wpcom_user' ) {
+						this.props.linkSocialUser( 'google', error.email );
 					}
 
 					this.recordEvent( 'calypso_login_social_login_failure', {
```

Comment:
My bad, this is already handled by the [`recordEvent()`](https://github.com/Automattic/wp-calypso/blob/4aadcc564e24c9d56c1a2f441c2a32fca852d8fa/client/blocks/login/social.jsx#L69) (even though it won't work very when we'll add a Facebook button).